cirq-core/cirq/ops/gate_features.py

```diff
@@ -69,14 +71,26 @@ def _num_qubits_(self) -> int:
         return 1
 
 
-class TwoQubitGate(raw_types.Gate, metaclass=abc.ABCMeta):
+class _TwoQubitGateMeta(value.ABCMetaImplementAnyOneOf):
+    def __instancecheck__(cls, instance):
+        return isinstance(instance, raw_types.Gate) and instance._num_qubits_() == 2
+
+
+@deprecated_class(deadline='v0.14', fix='Define _num_qubits_ manually.')
+class TwoQubitGate(raw_types.Gate, metaclass=_TwoQubitGateMeta):
```

Comment:
For (1), since we're removing TwoQubitGate from e.g. CX, we need to use the metaclass so that any user code like `isinstance(CX(), TwoQubitGate)` doesn't break.

For (2), IDK whether that would currently trigger a warning or whether that behavior is possible, but agree it would be best if it does. I'll look into it.

Reply:
Fixed